### PR TITLE
create a renderer tutorial part 1 + 2 added

### PR DESCRIPTION
**Pull request by LukeAnnison**
# Creating A Renderer Tutorial
- Part 1: Creating a Renderer
- Part 2: Adding Language Translation

**Future ideas**
- Part 3: Setting and connecting to a arprofile language config to detect and translate the language

### CODE REVIEW

**Comment by DanMacDonald** [comment]
Right off the bat, we really don't want to be teaching `Arweave Deploy` which has been deprecated for some time now.
https://github.com/ArweaveTeam/arweave-deploy

image

better to link to https://cookbook.arweave.dev/guides/deployment/arkb.html or https://cookbook.arweave.dev/guides/deployment/bundlr-cli.html

Personally I often just drag and drop files/directories with arweave.app

File: docs/src/guides/create-a-renderer/part-1-create-a-renderer.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+In this tutorial, you'll learn how to deploy a renderer on the Arweave network that displays text content fetched from the permaweb. Additionally, you will deploy a "Hello, World!" text file using Bundlr and render it using the deployed renderer.
+
+### Prerequisites
+
+Before we begin, make sure you have the following:
+
+-   Basic knowledge of JavaScript, HTML, and CSS
+-   An Arweave wallet with a small amount of AR to deploy the renderer and the text file
+-   [Arweave Deploy CLI](https://github.com/ArweaveTeam/arweave-deploy) installed and configured
```
Comment (DanMacDonald) [inline docs/src/guides/create-a-renderer/part-1-create-a-renderer.md:11]: Actually, I'm not sure why this is a prerequisite? It doesn't seem like it's used anywhere in the guide?

Comment (twilson63) [reply docs/src/guides/create-a-renderer/part-1-create-a-renderer.md:11]: Also you should not need AR to deploy a renderer if using bundlr-network

Comment (twilson63) [reply docs/src/guides/create-a-renderer/part-1-create-a-renderer.md:11]: I would recommend bundlr-network:

1 - its free for anything under 100k
2 - it is instant in terms of cache and comes with retry capability if it has trouble writing to the network
3 - it allows the network to scale 50k tps.

Comment (LukeAnnison) [reply docs/src/guides/create-a-renderer/part-1-create-a-renderer.md:11]: Brilliant. I thought as much, but wasn't sure if I was perhaps missing something RE reccomending the most basic atomic method for the docs. Updating the PR now

File: docs/src/guides/create-a-renderer/part-1-create-a-renderer.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
+## Part 1: Deploying a Renderer on Arweave
+
+In this tutorial, you'll learn how to deploy a renderer on the Arweave network that displays text content fetched from the permaweb. Additionally, you will deploy a "Hello, World!" text file using Bundlr and render it using the deployed renderer.
+
+### Prerequisites
+
+Before we begin, make sure you have the following:
+
+-   Basic knowledge of JavaScript, HTML, and CSS
+-   An Arweave wallet with a small amount of AR to deploy the renderer and the text file
```
Comment (twilson63) [inline docs/src/guides/create-a-renderer/part-1-create-a-renderer.md:10]: FYI, You don't need AR to deploy the renderer if under 100k

**Comment by twilson63** [comment]
@DanMacDonald 

Should we have a discussion, about how to introduce these concepts and guides?

Maybe at this weeks cookbook meetup?

**Comment by twilson63** [comment]
I kind of think these guides needs a concept reference in the cookbook:

What is a renderer? How it relates to Transactions, and reference to the draft spec, also a note that the standard is currently in draft mode and could change in the future.

🤔

**Comment by DanMacDonald** [comment]
Yeah, I keep editing the index of the cookbook, it's clear new contributors don't understand the various places a new document should be referenced.

**Comment by twilson63** [comment]
Maybe this should go under references/experimental?

**Comment by twilson63** [comment]
I think a guide for renderers now is a good thing, now that we have the Data Protocol Spec - Do you think it is worth refining this guide and publishing?

**Comment by atticusofsparta** [comment]
> I kind of think these guides needs a concept reference in the cookbook:
> 
> What is a renderer? How it relates to Transactions, and reference to the draft spec, also a note that the standard is currently in draft mode and could change in the future.
> 
> 🤔

@twilson63 @DanMacDonald just getting to this. We can add some eslint rules for formatting of the index and docs references i believe. Kind of a wierd use for it, but for example we can say any text that says "graphql" must be a link to the gql reference. Likewise for the index we can add formatter rules for alphabetizing and avoidance of page duplication i think. 

Think its worth it?

**Pull request closed without merging**